lib/solr/configuration.rb

```diff
@@ -1,16 +1,34 @@
+require 'solr/field'
+require 'solr/dynamic_field_type'
+require 'solr/field_definition_builder'
+require 'solr/type_definition_builder'
+
 module Solr
   class Configuration
-    attr_accessor :field_map, :open_timeout, :read_timeout, :url
+    attr_accessor :open_timeout, :read_timeout, :url, :field_types, :fields
```

Comment:
can we please make a single `faraday_options` config value that is a hash and has everything faraday needs? That way we'll greatly simplify future maintenance.